packages/core/src/components/modal/test/open-close/modal.e2e.ts

```diff
@@ -0,0 +1,55 @@
+import { test } from 'stencil-playwright';
+import { expect } from '@playwright/test';
+
+const componentTestPath = 'src/components/modal/test/open-close/index.html';
+
+test.describe('tds-modal-open', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(componentTestPath);
+  });
+
+  test('1. Modal is opened by default', async ({ page }) => {
+    const tdsModal = page.locator('tds-modal');
+    const hasShowAttribute = await tdsModal.getAttribute('show');
+    if (hasShowAttribute) {
+      await expect(tdsModal).toBeVisible(); // Modal is initially open
+      await expect(page).toHaveScreenshot({ maxDiffPixels: 0 });
+    }
+  });
+
+  test('Clicking [Delete] button closes the modal', async ({ page }) => {
+    const deleteButton = page.getByText('Delete');
+    await deleteButton.click(); // Click [Delete] button
+    const tdsModal = page.locator('tds-modal');
+    await expect(tdsModal).toBeHidden(); // Modal is closed
+  });
+
+  test('[Open Modal] button exists', async ({ page }) => {
+    const openModalButton = page.getByTestId('tds-modal-button-testid');
+    await expect(openModalButton).toBeVisible(); // [Open Modal] button exists and is visible
+  });
+
+  test('Clicking [Open Modal] button opens the modal', async ({ page }) => {
+    const openModalButton = page.getByTestId('tds-modal-button-testid');
+    const tdsModal = page.locator('tds-modal');
+
+    await openModalButton.dispatchEvent('click');
```

Comment:
After removing this click in the test it still comes back success. Because after clicking the button the modal should close right? Since its open by default.

I think this tests needs to be where the modal is not open by default to be tested. Also instead of getting the Open modal button like that since it would not be a <button> element we need to get the element inside tds-button with

```
const openModalButton = page.getByRole('button').filter({ hasText: /Open Modal/ });
```

Reply:
I'm not following you? Maybe the modal should be initially hidden? Wondering what should be the correct structure and approach. Maybe hidden by default and take a screenshot to verify? And then test to verify that it opens and one for closing.

Reply:
Yes I think "default" can be open by default.
"Open - Close" tests can be hidden by default and then does all the tests to check that the button opens and closes the modal.

Reply:
Yes, I think so too. Although, removing the show attribute and then trying to test that clickin the open modal button and that the modal should be visible, always fails?

`test('Clicking [Open Modal] button opens the modal', async ({ page }) => {
    const openModalButton = page.getByRole('button').filter({ hasText: /Open Modal/ });
    const tdsModal = page.locator('tds-modal');

    await openModalButton.click();
    await expect(tdsModal).toBeVisible(); // Modal should be open after clicking [Open Modal]
  });`
  
  Then I just get:
  
  ` Locator: locator('tds-modal')
    Expected: visible
    Received: `hidden`

Reply:
Yes the user don't know what a `page.locator('tds-modal')` so could either, take a screenshot to verify or to check for an element inside the modal to see if that is visible?

Im not 💯 why tds-modal is still hidden when its clearly should be visible. Mby it just dosnt know if it is somehow I havnt tested it myself.

Reply:
It works for me with the above code if I add the selector="#my-modal-button" to the code

Reply:
> Did you add the selector to the modal in the index.html? you can also open the index.html on `Live Server` extension to see it and test it.
> 
> Edit: It works for me with the above code if I add the selector="#my-modal-button" to the code

Ah yea? Hm, I have it in index.html, but still it fails?

Reply:
Hmm strange, Then I don't know :(

Reply:
<tds-modal needs to have `selector` prop that targets button's ID. I can see on my local that it is missing.